spec/unit/iiif/v3/presentation/service_spec.rb

```diff
@@ -1,28 +1,150 @@
 describe IIIF::V3::Presentation::Service do
 
-  describe 'self#get_descendant_class_by_jld_type' do
-    before do
-      class DummyClass < IIIF::V3::Presentation::Service
-        TYPE = "Collection"
-        def self.singleton_class?
-          true
-        end
+  describe '#required_keys' do
+    it '"type" is not required' do
+      expect(subject.required_keys).not_to include('type')
+    end
+  end
+
+  describe '#prohibited_keys' do
+    keys = IIIF::V3::Presentation::Service::CONTENT_RESOURCE_PROPERTIES +
+      IIIF::V3::Presentation::Service::PAGING_PROPERTIES +
+      %w{
+        nav_date
+        viewing_direction
+        start_canvas
+        content_annotation
+      }
+    keys.each do |k|
```

Comment:
a slightly more concise alternative would be:
```ruby
it 'contains the expected key names' do
  expect(subject.prohibited_keys).to include(*keys)
end
```

which i slightly prefer because it makes this whole thing count as one test, instead of a test for each element.  but i would have no problem merging with this part as-is, whether it's because it's a hassle to change it, or because you prefer the current version regardless.  test failure output reads the same either way (i just tried on my laptop).